### ISSUE TITLE
theron simulation not working

### ISSUE THREAD

**Issue by mariadelmar2497** [RobotnikAutomation/rb_theron_sim#1]
Hi, I'm trying to deploy the rb_theron simulation, but it is still not working. I have followed all the steps in the Readme, but when I launch the complete simulation, the following error occurs:
![image](https://user-images.githubusercontent.com/113019574/188840612-55672b92-8fff-4677-a820-779276145cd7.png)

**Comment by RomanRobotnik** [comment]
Hi,

It seems you are using ROS Noetic, right? The simulation is not supported by ROS Noetic, or at least it has not been tested. It could be related to changes in the xacro format.

**Comment by alex-arnal** [comment]
Hi @mariadelmar2497 ,

you can try commenting the macro named box_intertia_robot (from line 5 to 9) in theron_base.urdf.xacro. It seems to be the source of the error and from what I have reviewed, it should not be necessary in this file.

https://github.com/RobotnikAutomation/rb_theron_common/blob/0cae87e57b63c0dc5c59aed501fdcc7ad9c40f78/rb_theron_description/urdf/bases/theron_base.urdf.xacro#L5:L9

**Comment by mariadelmar2497** [comment]
Thank you! Now the theron simulation is running. But it seems the laser is not reading well, so the robot localization is not working. 

![Screenshot from 2022-10-14 11-20-49](https://user-images.githubusercontent.com/113019574/195812381-da200a63-166e-4f78-b55e-f1296ed3b21f.png)

![Screenshot from 2022-10-14 11-23-01](https://user-images.githubusercontent.com/113019574/195812408-0c0d80f3-c6ae-4d22-affe-a3b9eaa88b47.png)

**Comment by alex-arnal** [comment]
Hi @mariadelmar2497 ,

Maybe this error is due to some problem related to the plugin used by gazebo to simulate the laser readings.

You can try to install the plugin if you have not already installed:
```bash
sudo apt-get install ros-melodic-gazebo-plugins
```

If the issue persist, I recommend you to update RB-THERON packages (sim and common) with last changes I pushed to allow gazebo simulate the laser readings but no using the gpu plugin. After updating the packages you will need to set the argument use_gpu to false when launching the simualtion:
```bash
roslaunch rb_theron_sim_bringup rb_theron_complete.launch use_gpu:=false
```

I hope one of the options works for you.

Álex.

**Comment by mariadelmar2497** [comment]
Hello,

I did what you commented, but now this error occurs:

![captura](https://user-images.githubusercontent.com/113019574/197169505-ce306c08-7527-4608-a3a4-8e52a99a08e4.png)

If you can provide a requirements.txt file, I can check which version I need from libraries and Gazebo.

Thank you so much

**Comment by alex-arnal** [comment]
Hi @mariadelmar2497 ,

it would be really helpful if you provide the output of the terminal when running the command (correct me if I'm wrong)

```bash
roslaunch rb_theron_sim_bringup rb_theron_complete.launch use_gpu:=false
```

Maybe we can find out what is going on now.

Thanks,
Álex.

**Comment by mariadelmar2497** [comment]
Hi again,

I think my problem was with Gazebo libraries, because I have upgraded Gazebo9 to the latest version and I have installed again some libraries and it seems that the laser is working now.I am testing the simulation and seems to work fine, despite sometimes when the robot randomly thinks is locked and Gazebo simulation collapses and restart at the origin.

![captura2](https://user-images.githubusercontent.com/113019574/197175006-552f911d-ae98-4f1a-821c-56fdec4563aa.png)
![Captura de pantalla de 2022-10-21 12-28-25](https://user-images.githubusercontent.com/113019574/197175183-d126c6ce-b87d-4c94-b7ae-71d004f205f0.png)

Maybe because this map is a bit complicated to trajectory optimization algorithm, with many narrow corridors. 

Thank you so much!
